### DISCUSSION TITLE
Integration with Zotero

### DISCUSSION THREAD

**Discussion by sahil48** [ahrm/sioyek#363] [Ideas]
I've been finding myself using Zotero more to store and manage my PDF files. Accessing the files from Sioyek would be nice.

**Comment by romor001** [comment]
Well, you can set Sioyek as the program to open files in Zotero. This works nicely.

But the Zotfile extension of Zotero will not be able to detect the highlights and annotations you'll make within Sioyek. At least for me the _embed_annotations extension of Sioyek does not do the trick. 
Is there anything one has to do with that extension to actually save the file with the highlights embedded?

**Reply by ahrm** [reply]
> At least for me the _embed_annotations extension of Sioyek does not do the trick.

What happens when you run the extension from the command line? That is, running
```
python -m sioyek.embed_annotations "%{sioyek_path}" "%{local_database}" "%{shared_database}" "%{file_path}"
```
After replacing `%{}`s with the correct paths.

**Comment by anandkumar89** [comment]
Not sure if its totally wild, but it it were possible to write sioyek highlights to zotero database, would be helpful. This would avoid requiring opening zotero. not necessarily through app, but even via a plugin.

Also, would it be possible to add highlights from zotero database.

I am willing to try and hack but have no experience writing plugins for either, just know some python and could work with databases.

**Reply by Mikilio** [reply]
After trying the above solution for a few days, I really find myself wanting this, actually. I would like to bring all highlights of zotero into sioyek via a different command like:

zotero_highlight_text
zotero_highlight_box
zotero_sticky_note

That could make an amazing plugin.

**Comment by Mikilio** [comment]
So what is currently the best way to get my highlights into zotero?

**Reply by anandkumar89** [reply]
https://github.com/ahrm/sioyek/discussions/363#discussioncomment-5280265 this. Have not tried it yet!

**Comment by ryanwwest** [comment]
It would be so cool if Sioyek could be integrated into the Zotero reader as an add-on (also probably very infeasible, but both systems do both support Windows+macOS+Linux). Zotero reads PDFs but lacks all of Sioyek's cool features and keyboard navigation, rivaling it only with annotation support. Zotero and Sioyek are both aimed at research papers so they seem like a natural fit. (Though @ahrm, I'm curious if many more features are being planned to add to Sioyek or if it's mostly feature-complete/bug fixes only at this point for you)
